lib/utils/addressUtils.ts

```diff
@@ -0,0 +1,32 @@
+import { Socket } from 'net';
+import { Address } from '../types/p2p';
+import assert from 'assert';
+
+/** Helper methods for interacting with the [[Address]] type. */
+export const addressUtils = {
```

Comment:
This should be exported as the default.

Besides that, I preferred the class implementation, as it was before, specially because `toString` is called many times. In my earlier comment I was just wandering about the naming, but it wasn't an issue. 

Both solutions are fine.

Reply:
The reason I wanted a type was for the `NodeFactory` type definition plus a handful of other places where we were using host+port tuples. Using `SocketAddress` wouldn't have worked I believe because it was a class with a public `toString` method. I agree the current `toString` usage is a bit awkward, but if it's ok with you then we can leave it.